SelfAndPeerEvaluations/partner_eval_quiz_creator.py

```diff
@@ -0,0 +1,161 @@
+
+            quiz_questions.append(question)
+        return quiz_questions
+
+    def __create_identify_partner_answers(self) -> dict:
+        """
+        Creates the multiple choice answers for "Who is your partner?" question
+        If a student has the same sortable_name as another student, then we
+        try to see if the student's name is unique with the last 4 digits of their SID appended (no SID -> XXXXXXXXX appended).
+        If a student still isn't unique with their last 4 SID digits append, then we try their last 5 SID digits.
+        It is extremely unlikely (at least 1 in 9^5 chance) that there exists students with
+        the same sortable_name AND same last 5 digits.
+        :return: dict of answers properly formated for canvasapi
+        """
+        students = make_student_id_map(self.course).keys()
+        answers = [{"answer_text": student, "answer_weight": 1} for student in students]
+        answers.append(
+            {"answer_text": "I did not submit with a partner.", "answer_weight": 1}
+        )
+        return dict(enumerate(answers))
+
+    def __find_partner_eval_ag_id(self, course: canvasapi.course.Course) -> int:
+        partner_eval_ag = None
+
+        for ag in course.get_assignment_groups():
+            # FIXME: this is too strict
+            #        we should be able to match similar assignment groups
+            #        or, allow user to create their own ag
+            if ag.name == "Partner Evalulations":
+                partner_eval_ag = ag
+
+        if partner_eval_ag is None:
+            print('"Partner Evaluations" category does not exist!')
+            print('Creating "Partner Evalutions" assignment group')
+            partner_eval_ag = course.create_assignment_group(
+                name="Partner Evalulations", group_weight=10
+            )
+        else:
+            print("Not making a new assignment group since ")
+            print('"Partner Evaluations" assignment group exists.')
+
+        return partner_eval_ag.id
+
+
+if __name__ == "__main__":
+    url = "https://canvas.ucdavis.edu"
+    key = str(input("Enter API key:"))
+
+    canvas = canvasapi.Canvas(url, key)
+    course = canvas.get_course(1599)  # sandbox course ID
+
+    with open("./self_and_partner_evaluation_questions.json") as f:
+        json_questions = json.load(f)
+
+    # first_due = datetime.datetime(2022, 7, 20, 23, 59)
+    # first_open = datetime.datetime(2022, 7, 20, 19, 00)
+    assignment_names = ["json quiz x"]
+    for _ in assignment_names:
+        # due = first_due + datetime.timedelta(weeks_into_the_quarter * 7)
+        # close = due + datetime.timedelta(days=1)
+        # open_d = first_open + datetime.timedelta(weeks_into_the_quarter * 7)
+        # quiz = PartnerEvalQuizCreator(course, _, open_d, due, close)
+        quiz = PartnerEvalQuizCreator(course, json_questions, _)
+        quiz.upload_to_canvas()
+        # weeks_into_the_quarter += 1
```

Comment:
Need to add a full fledged piece of code for collecting values from users to input our program.